services/core/functions/add-repo/index.ts

```diff
@@ -1,53 +1,109 @@
+
+type AddRepoEventBody = {
+  detail: {
+    repositories : [[{
+      name: string,
+      full_name: string,
+    }]],
+    sender: {
+      login: string,
+      id: string,
+    }
+  }
 };
 
-const isValidEventBody = (input: unknown): input is AddRepoLambdaInput =>
-  typeof input === "object" &&
-  input !== null &&
-  "reposAdded" in input &&
-  typeof input.reposAdded === "string" &&
-  "userId" in input &&
-  typeof input.userId === "string";
+type User = {
+  userId : string,
+  repos: string[],
+}
 
-export const main = async (
-  event: APIGatewayProxyEvent
+type Repo = {
+  name: string,
+  full_name: string,
+}
+
+export const main = async(
+  event: AddRepoEvent
 ): Promise<FormattedHandlerResponse> => {
-  if (event.body === null) {
-    return formatResponse(
-      "The request does not contain a body as expected.",
-      400
-    );
-  }
+  const eventBody = event as unknown as AddRepoEventBody;
 
   try {
-    const inputBody: unknown = JSON.parse(event.body);
+    const repositories = eventBody.detail.repositories as unknown as [Repo];
+    const sender = eventBody.detail.sender;
+
+    const response = await AuthEntity.query("ACCOUNT#github", {
+      index: "GSI1",
+      eq: `ACCOUNT#${sender.id}`,
+      attributes: [ "userId" ],
+    });
+
+    if (!response.Items || response.Items.length === 0) {
+      return formatResponse("Error, no user found with corresponding github id.", 204);
+    }
 
-    if (!isValidEventBody(inputBody)) {
-      return formatResponse(
-        "The request body does not contain the expected data.",
-        400
+    const githubUser = response.Items[0] as User;
+
+    const userResponse = await UserEntity.get({
+      userId: githubUser.userId,
+    });
+
+    let hasRepos = false;
+    if (userResponse.Item && "repos" in userResponse.Item) {
+      hasRepos = true;
+    }
+
+    const user = userResponse.Item as unknown as User;
+
+    let newRepos: string[];
+    if (hasRepos) {
+      newRepos = repositories.filter((repo) =>
+        !user.repos.includes(repo.name)
+      ).map((repo) => repo.name);
+    }
+    else {
+      newRepos = repositories.map((repo) => repo.name);
+    }
+
+    let allRepos: string[];
+    if (hasRepos) {
+      allRepos = user.repos.concat(
+        newRepos
       );
     }
+    else {
+      allRepos = newRepos;
+    }
 
     await UserEntity.update(
       {
-        userId: inputBody.userId,
-        repos: inputBody.reposAdded,
+        userId: githubUser.userId,
+        repos: allRepos,
       },
       { conditions: { attr: "userId", exists: true } }
     );
 
-    return formatResponse("User updated successfully.");
+    await Promise.all(newRepos.map(async (repo) => {
+      const repoId = uuidv4();
+      await RepoEntity.put({
+        repoId: repoId,
+        prompt: instructionPrompt,
+        name: repo,
+        ownerId: githubUser.userId,
+      });
+    }));
+    
+    return formatResponse("User repos updated successfully.", 200);
```

Comment:
Since this lambda is no longer being invoked by the api Gateway, we don't need to return these responses. We can probaly just have the function return void, and maybe add some logging ?